lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -1611,3 +1630,25 @@ pub(crate) struct EmptyYieldVal;
 fn default_fatal_handler(inst: &Instance) -> ! {
     panic!("> instance {:p} had fatal error: {}", inst, inst.state);
 }
+
+/// Used by hosts to limit resource consumption of instances.
+///
+/// An instance can be configured with a resource limiter so that hosts can take into account non-WebAssembly
+/// resource usage to determine if linear memory should be permitted to grow.
+pub trait ResourceLimiter {
+    /// Called when the the instance's linear memory has been requested to grow.
+    ///
+    /// * `current` gives the current size of the linear memory, in bytes.
+    /// * `desired` gives the desired new size of the linear memory, in bytes.
+    /// * `limit` gives the configured maximum size of the linear memory, in bytes.
+    ///
+    /// Returning `Ok` signals that the memory should be permitted to grow; returning an `Err(err)` prevents
+    /// the growth from occurring, yielding an `Error::LimitsExceeded(err)` from the call to grow the memory.
+    fn memory_growing(
+        &self,
+        instance: &Instance,
+        current: u32,
+        desired: u32,
+        limit: u32,
+    ) -> Result<(), String>;
```

Comment:
`String` feels a bit weird here as an error value, I think because I have a hard time imagining we'd want a call to `memory_growing` to fail with any limit description other than "linear memory". Between this and the above comment about the `limit` argument I'm also wondering if I might just be thinking about this trait incorrectly :slightly_smiling_face: